scalafix-tests/output/src/main/scala/tests/ExplicitResultTypesImports.scala

```diff
@@ -18,8 +18,7 @@ object ExplicitResultTypesImports {
 
   val timezone: ju.TimeZone = null.asInstanceOf[java.util.TimeZone]
 
-  // TODO: Is this desirable behavior?
-  val inner: Searching.SearchResult = null.asInstanceOf[scala.collection.Searching.SearchResult]
+  val inner: SearchResult = null.asInstanceOf[scala.collection.Searching.SearchResult]
```

Comment:
The primary motivation for the old behavior comes from the desire to insert `Term.Name` and `import scala.meta.Term` instead of `import scala.meta.Term.Name`. Scalameta isn't the best motivating example, but I feel like this pattern of nesting generic names inside a class is not only limited to Scalameta. Any thoughts on how to support both use-cases? I'm open for various routes

* default to importing FQN with option to flip default
* ability to manually include/exclude individual prefixes from nesting

Reply:
Thanks for feedback, I didn't realize the current behavior was an active choice. It makes sense with your example, I am fine keeping it, this was just a side effect I got when iterating on my own test. I have amended the last commit to keep/document the current behavior and align it with path-dependent types, see https://github.com/scalacenter/scalafix/pull/1276/commits/fffba0667b7e72767e58f360e8e245b0999fb851.